engine/main.py

```diff
@@ -640,6 +644,18 @@ def unsubscribe_group(group_name, user):
     logging.debug(res)
     return res
 
+def confirm_unsubscribe(membergroups, user):
```

Comment:
Maybe it membergroup not plural. 
Can you add a block comment under the function definition? e.g.,
    """This function is blah blah
        Args:
            membergroup (MemberGroup): a MemberGroup instance of a user who tries to unsubscribe
            user (blah):
    """